internal/tagencoding/tagencoding.go

```diff
@@ -38,14 +32,11 @@ func (vb *Values) growIfRequired(expected int) {
 }
 
 func (vb *Values) WriteValue(v []byte) {
-	length := len(v)
-	vb.growIfRequired(sizeOfUint16 + length)
+	length := len(v) & 0xff
```

Comment:
Did we instead mean to use `0xffff` uint16 and not uint8?

Reply:
I changed the encoding from uint16 to uint8, since we restrict tag values to length 255 now.

(Which I remembered when I tried to test this code with values longer than 255.  :))

Reply:
Oops I missed the description and jumped straight into code, @rakyll corrected me offline. My apologies.

Reply:
@jcd2 gotcha gotcha, thanks for the PR.